tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3830,6 +3830,37 @@ public function theResponseShouldContain(string $method, string $user, string $s
 		}
 	}
 
+	/**
+	 * @Then as user :user the key :key from PROPFIND response should match with sharedWithMe driveItemId of share :resource
+	 *
+	 * @param string $user
+	 * @param string $key
+	 * @param string $resource
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function asUserTheKeyFromPropfindResponseShouldMatchWithSharedwithmeDriveitemidOfShare(string $user, string $key, string $resource): void {
+		$xmlRes = $this->featureContext->getResponseXml();
+		$fileId = $xmlRes->xpath("//d:response/d:propstat/d:prop/oc:name[text()='$resource']/preceding-sibling::$key")[0]->__toString();
+
+		$jsonResponse = GraphHelper::getSharesSharedWithMe(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$user,
+			$this->featureContext->getPasswordForUser($user)
+		);
+		$jsonResponseBody = $this->featureContext->getJsonDecodedResponseBodyContent($jsonResponse);
+		foreach ($jsonResponseBody->value as $value) {
+			if ($value->name === "$resource") {
+				$driveItemId = $value->id;
+			} else {
+				throw new Error("Response didn't contain a share $resource");
+			}
+		}
+		Assert::assertEquals($fileId, $driveItemId, "Fileid '$fileId' doesn't match driveItemId '$driveItemId'");
```

Comment:
```suggestion
		Assert::assertEquals($fileId, $driveItemId, "Fileid '$fileId' doesn't match driveItemId '$driveItemId'");
```
Either File-id or FileId